### PR TITLE
add checksum verification for downloaded archive

### LINKED ISSUE #67: checksum verification for archive download
please add checksum verification for archive downloads:

### PR DESCRIPTION
**Pull request by hex2a**
##### SUMMARY
This PR adds checksum verification for the archive install method.

Closes #67 
##### TESTS/SPECS
Travis CI tests are passing.

It was tested in a Debian 8 environment.

### CODE REVIEW

File: manifests/params.pp [hunk 1]
```diff
@@ -20,6 +20,25 @@
   $package_name       = 'vault'
   $package_ensure     = 'installed'
 
+  $checksum_type       = 'sha256'
+  $checksums           = {
```
Comment (jsok) [inline manifests/params.pp:24]: This is going to be a big maintenance burden, every version bump is going to require me to lookup all these checksums.

**Review by jsok — CHANGES_REQUESTED** [review]
A user should be able to specify their own checksum in the scenario of running a custom or RC build. So only allowing them to choose from a pre-define list is too restrictive.

Will also require proper rspec tests for any PR to be acceptable.

File: manifests/params.pp (hunk 1 above)
Comment (jsok) [reply manifests/params.pp:24]: I'd suggest writing a custom `rake` task which fetched all the checksums for a given official vault release.

Comment (hex2a) [reply manifests/params.pp:24]: would this rake test be executed while building the module?

Comment (jsok) [reply manifests/params.pp:24]: No it's more of a helper/maintenance thing, even if it just writes the checksums to stdout. Just imagine you need to bump the vault version, how do you now also update the checksums?

**Comment by hex2a** [comment]
The checksum will be looked up using "$os_$arch" in the hash, a user can set their own checksum by setting it for the appropiate key.

File: manifests/params.pp (hunk 1 above)
Comment (hex2a) [reply manifests/params.pp:24]: it should just output the checksums as puppet hash, so it can be pasted in the params file?

**Comment by jsok** [comment]
It's a little non-idiomatic puppet to provide a parameter that way IMO. The `$checksums` hash is just an internal detail, the user should either say:

1. Use no checksum
1. Use the default release one
1. Use this specific one

Asking them to construct a hash like that is a little mean.

File: manifests/params.pp (hunk 1 above)
Comment (jsok) [reply manifests/params.pp:24]: That would be a good start, yes.
Currently I manually version bump but something which could one day be automated.

**Comment by hex2a** [comment]
As for the tests, to you have any suggestion how to test this?

I'd rather like to write an acceptance test for it, however they currently are not working for me. (#65)

**Comment by jsok** [comment]
If we assume the `archive` module works, all the specs need to test is that we marshall the checksum args correctly.

I've created #65 to get acceptance tests going again.

**Pull request closed without merging**